### PR TITLE
[pipelines] readiness and liveness probe configurations

### PR DESCRIPTION
**Pull request by stein321**
#### PR Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Chart Version bumped
- [x] CHANGELOG.md updated
- [ ] Variables and other changes are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[artifactory]`)

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by peters95** [comment]
The defaults in values.yaml are different than the default in the template. Is there a reason for this?

**Comment by stein321** [comment]
@peters95 my mistake. updated them to match

**Review by peters95 — APPROVED** [review]
LGTM

File: stable/pipelines/templates/pipelines-statefulset.yaml [hunk 1]
```diff
@@ -246,16 +250,20 @@ spec:
...
-            failureThreshold: 6
-            timeoutSeconds: 5
+            initialDelaySeconds: {{ default 10 .Values.pipelines.www.livenessProbe.initialDelaySeconds }}
+            timeoutSeconds: {{ default 5 .Values.pipelines.www.livenessProbe.timeoutSeconds }}
+            periodSeconds: {{ default 5 .Values.pipelines.www.livenessProbe.timeoutSeconds }}
+            failureThreshold: {{ default 6 .Values.pipelines.www.livenessProbe.failureThreshold }}
+            successThreshold: {{ default 1 .Values.pipelines.www.livenessProbe.successThreshold }}
           readinessProbe:
             httpGet:
               path: /
```
Comment (chukka) [inline stable/pipelines/templates/pipelines-statefulset.yaml:260]: @stein321 Can we also move `path` and `port` to values.yaml

File: stable/pipelines/values.yaml [hunk 2]
```diff
@@ -131,7 +131,19 @@ pipelines:
       ## Whitelist IPs allowed to LoadBalancer type services
       ## Example: loadBalancerSourceRanges={82.82.190.51/32,141.141.8.8/32}
       loadBalancerSourceRanges: []
-
+    livenessProbe:
```
Comment (chukka) [inline stable/pipelines/values.yaml:134]: Also, can we have switch to enable/disable to  livenessProbe or readinessProbe , please refer [artifactory chart](https://github.com/jfrog/charts/blob/master/stable/artifactory/values.yaml#L511) for an example

File: stable/pipelines/templates/pipelines-statefulset.yaml (hunk 1 above)
Comment (stein321) [reply stable/pipelines/templates/pipelines-statefulset.yaml]: resolved

**Review by chukka — APPROVED** [review]

File: stable/pipelines/values.yaml [hunk 3]
```diff
@@ -131,6 +131,25 @@ pipelines:
...
       loadBalancerSourceRanges: []
+    livenessProbe:
+      enabled: true
+      initialDelaySeconds: 20
+      timeoutSeconds: 10
+      periodSeconds: 10
+      failureThreshold: 10
+      successThreshold: 1
+      path: /
+      port: api
```
Comment (chukka) [inline stable/pipelines/values.yaml:142]: @stein321 is `api` correct value for port ?

Comment (stein321) [reply stable/pipelines/values.yaml:142]: yes. that's what it's set in master anyway

**Review by eldada — APPROVED** [review]

File: stable/pipelines/templates/pipelines-statefulset.yaml [hunk 4]
```diff
@@ -212,20 +212,29 @@ spec:
...
             httpGet:
-              path: /
-              port: api
-            initialDelaySeconds: 10
-            timeoutSeconds: 5
-            failureThreshold: 6
+              path: {{ default "/" .Values.pipelines.api.livenessProbe.path}}
+              port: {{ default "api" .Values.pipelines.api.livenessProbe.port}}
+            initialDelaySeconds: {{ default 20 .Values.pipelines.api.livenessProbe.initialDelaySeconds }}
+            timeoutSeconds: {{ default 10 .Values.pipelines.api.livenessProbe.timeoutSeconds }}
```
Comment (danielezer) [inline stable/pipelines/templates/pipelines-statefulset.yaml:222]: Remove the use of default here. Place the default values in values.yaml

**Review by danielezer — CHANGES_REQUESTED** [review]

File: stable/pipelines/templates/pipelines-statefulset.yaml (hunk 4 above)
Comment (eldada) [reply stable/pipelines/templates/pipelines-statefulset.yaml:222]: +1

Comment (stein321) [reply stable/pipelines/templates/pipelines-statefulset.yaml]: done

**Review by danielezer — APPROVED** [review]

**Pull request merged**